spec/jobs/email_csv_job_spec.rb

```diff
@@ -39,18 +40,44 @@
       allow(storage_service).to receive(:write_file)
       allow(storage_service).to receive(:get_presigned_url).and_return(test_url)
 
+      allow(Csv::SalesLogCsvService).to receive(:new).and_return(sales_log_csv_service)
+      allow(sales_log_csv_service).to receive(:prepare_csv).and_return("")
+
       allow(CsvDownloadMailer).to receive(:new).and_return(mailer)
       allow(mailer).to receive(:send_csv_download_mail)
     end
 
-    it "uses an appropriate filename in S3" do
-      expect(storage_service).to receive(:write_file).with(/logs-.*\.csv/, anything)
-      job.perform(user)
+    context "when exporting lettings logs" do
+      it "uses an appropriate filename in S3" do
+        expect(storage_service).to receive(:write_file).with(/lettings-logs-.*\.csv/, anything)
+        job.perform(user)
+      end
+
+      it "includes the organisation name in the filename when one is provided" do
+        expect(storage_service).to receive(:write_file).with(/lettings-logs-#{organisation.name}-.*\.csv/, anything)
+        job.perform(user, nil, {}, nil, organisation)
+      end
     end
 
-    it "includes the organisation name in the filename when one is provided" do
-      expect(storage_service).to receive(:write_file).with(/logs-#{organisation.name}-.*\.csv/, anything)
-      job.perform(user, nil, {}, nil, organisation)
+    context "when exporting sales logs" do
+      it "uses an appropriate filename in S3" do
+        expect(storage_service).to receive(:write_file).with(/sales-logs-.*\.csv/, anything)
+        job.perform(user, nil, {}, nil, nil, nil, "sales")
+      end
+
+      it "includes the organisation name in the filename when one is provided" do
+        expect(storage_service).to receive(:write_file).with(/sales-logs-#{organisation.name}-.*\.csv/, anything)
+        job.perform(user, nil, {}, nil, organisation, nil, "sales")
+      end
+
+      it "creates a SalesLogCsvService with the correct export type" do
+        expect(Csv::SalesLogCsvService).to receive(:new).with(export_type: "labels")
+        codes_only = false
+        job.perform(user, nil, {}, nil, nil, codes_only, "sales")
+        expect(Csv::SalesLogCsvService).to receive(:new).with(export_type: "codes")
```

Comment:
should these be separate tests?

Reply:
I don't think it really matters. Happy to be convinced otherwise